### PR TITLE
Vantuz lab3

### PR DESCRIPTION
**Pull request by ivanko337**

### CODE REVIEW

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Blogger.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class Blogger : Worker
+    {
+        public Blogger(int salary) : base(salary)
+        { }
+
+        public void CheckPayment()
+        { }
+
+        public void SlapInPassager(Passager passager)
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Blogger.cs:17]: Passenger)

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Blogger.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Blogger.cs:3]: remove unused imports

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Bus.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace FIrstUMLLab
+{
+    class Bus : GroundTransport
+    {
+        public void FillFuel()
+        { }
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Bus.cs:6]: bus cannot fill fuel itself

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Driver.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Driver.cs:5]: remove unused imports

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Driver.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class Driver : Worker
+    {
+        public Driver(int salary) : base(salary)
+        { }
+
+        public void Drive()
+        { }
+
+        public void FillFuel()
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Driver.cs:17]: pass Vehicle as parameter here

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Electrobus.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿namespace FIrstUMLLab
+{
+    class Electrobus : Bus
+    {
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Electrobus.cs:4]: add constructor at least...

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FIrstUMLLab
+{
+    public class GroundTransport : Transport
+    {
+        public List<Blogger> Bloggers { get; set; }
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs:11]: do you mean that bloggers will be at vehicle all the way? if not, remove it

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        public Validator Validator { get; private set; }
+
+        public GroundTransport()
+        {
+            this.Validator = new Validator();
+        }
+
+        public GroundTransport(Driver driver, List<DateTime> timetable, int amount, int stock)
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs:19]: add these params as class fields

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Passager.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        public bool IsPayed { get; private set; } = false;
+        public Transport UseTransport { get; private set; }
+
+        public Passager(Transport transport)
+        {
+            this.UseTransport = transport;
+        }
+
+        public void Payed()
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Passager.cs:19]: what is responsibility of that method?

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Passager.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+using System.Threading.Tasks;
+
+namespace FIrstUMLLab
+{
+    public class Passager : Person
+    {
+        public bool IsPayed { get; private set; } = false;
+        public Transport UseTransport { get; private set; }
+
+        public Passager(Transport transport)
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Passager.cs:14]: passenger will not always be in the transport, so remove that param and add base call at least

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Subway.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿namespace FIrstUMLLab
+{
+    class Subway : UndergroundTransport
+    {
+        public Turnstiles Turnstiles { get; private set; }
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Subway.cs:5]: plural form means that you have to declare List<Turnstile>, do not complicate your code

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Transport.cs [hunk 12]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace FIrstUMLLab
+{
+    public class Transport
+    {
+        public List<DateTime> Timetable { get; private set; }
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Transport.cs:8]: may be Dictionary<DateTime, Station> will be more applicable here?

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Worker.cs [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FIrstUMLLab
+{
+    public class Worker : Person
+    {
+        public int Salary { get; private set; }
+
+        public Worker(int salary)
```
Comment (secourse2019) [inline UML_vantuz/FIrstUMLLab/FIrstUMLLab/Worker.cs:13]: how about base call?

**Review by secourse2019 — CHANGES_REQUESTED** [review]
I'd like to see only source files (how to fix it, see Kate's PR)

too many duplicates

fix comments

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Driver.cs (hunk 5 above)
Comment (ivanko337) [reply UML_vantuz/FIrstUMLLab/FIrstUMLLab/Driver.cs:17]: > 
> 
> pass Vehicle as parameter here

it's redundant, Transport class have property TransportDriver

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs (hunk 7 above)
Comment (ivanko337) [reply UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs:11]: > 
> 
> do you mean that bloggers will be at vehicle all the way? if not, remove it

no, that code able to add and remove bloggers from vehicle

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Transport.cs (hunk 12 above)
Comment (ivanko337) [reply UML_vantuz/FIrstUMLLab/FIrstUMLLab/Transport.cs:8]: > 
> 
> may be Dictionary<DateTime, Station> will be more applicable here?

if comeone need to check timetable, he can print all timetable. like at transport stations

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs (hunk 7 above)
Comment (secourse2019) [reply UML_vantuz/FIrstUMLLab/FIrstUMLLab/GroundTransport.cs:11]: ok

File: UML_vantuz/FIrstUMLLab/FIrstUMLLab/Transport.cs (hunk 12 above)
Comment (secourse2019) [reply UML_vantuz/FIrstUMLLab/FIrstUMLLab/Transport.cs:8]: you can still print the keys from dict

**Review by secourse2019 — APPROVED** [review]

**Pull request merged**